syncer/syncer.go

```diff
@@ -971,6 +998,10 @@ func (s *Syncer) addJob(job *job) error {
 		queueBucket = s.cfg.WorkerCount
 		startTime := time.Now()
 		s.jobs[queueBucket] <- job
+		if waitXIDStatus(s.waitXIDJob.Load()) == waiting {
```

Comment:
If we are executing a DDL when we receive a `<-ctx.Done()`, the current DDL will definitely be canceled after this PR.

Reply:
IMO this is expected behaviour because DDL may cost a long time. Do you mean we should wait 10s?

Reply:
Maybe. I'm afraid some DDLs are not reentrant. But since we have ignored some database/table/index/column exists/notExists errors, it may also be fine.